### PR TITLE
Implement trace sink service

### PR DESCRIPTION
**Pull request by Jim3Things**
Adds the trace sink service implementation, unit tests
Remove LocalProduction, replace with use of SpanKindInternal as an indicator of an infrastructure-only span
Remove duplicate trace formatting code
Linter fixes

### CODE REVIEW

File: internal/services/frontend/session_manager.go [hunk 1]
```diff
@@ -76,7 +75,7 @@ func removeSession(session *sessions.Session) {
 	purgeStaleSessions()
 
 	if id, ok := session.Values[sessionIDKey].(int64); ok {
-		if entry, ok := activeSessions[id]; ok {
+		if entry, ok2 := activeSessions[id]; ok2 {
```
Comment (MikeCJohnsonUk) [inline internal/services/frontend/session_manager.go:78]: Is there any reason the outer "ok" needs to be retained?

File: internal/services/frontend/session_manager.go [hunk 2]
```diff
@@ -88,14 +87,14 @@ func removeSession(session *sessions.Session) {
...
-func getSession(session *sessions.Session) (SessionState, bool) {
+func getSession(session *sessions.Session) (sessionState, bool) {
 	mutex.Lock()
 	defer mutex.Unlock()
 
 	purgeStaleSessions()
 
 	if id, ok := session.Values[sessionIDKey].(int64); ok {
-		if entry, ok := activeSessions[id]; ok {
+		if entry, ok2 := activeSessions[id]; ok2 {
```
Comment (MikeCJohnsonUk) [inline internal/services/frontend/session_manager.go:97]: Is there any reason the outer "ok" needs to be retained?

File: internal/services/frontend/users.go [hunk 3]
```diff
@@ -151,8 +152,14 @@ func handlerUserCreate(w http.ResponseWriter, r *http.Request) {
 
 		w.Header().Set("ETag", fmt.Sprintf("%v", rev))
 
-		st.Infof(ctx, tick(), "Created user %q, pwd: <redacted>, enabled: %v, accountManager: %v", username, u.Enabled, u.CanManageAccounts)
-		_, err = fmt.Fprintf(w, "User %q created.  enabled: %v, can manage accounts: %v", username, u.Enabled, u.CanManageAccounts)
+		st.Infof(
+			ctx, common.Tick(),
+			"Created user %q, pwd: <redacted>, enabled: %v, accountManager: %v",
+			username, u.Enabled, u.CanManageAccounts)
+		_, err = fmt.Fprintf(
```
Comment (MikeCJohnsonUk) [inline internal/services/frontend/users.go:159]: An intervening blank line perhaps? I was trying to figure out why the fmt.Fprintf() was part of the st.Infof() 'til the penny dropped.

File: internal/tracing/exporters/common/common.go [hunk 4]
```diff
@@ -61,7 +66,44 @@ func ExtractEntry(_ context.Context, data *trace.SpanData) *log.Entry {
...
+		entry.GetParentID(),
+		deferredFlag(deferred),
+		infraFlag(entry.GetInfrastructure()),
+		entry.GetStatus(),
+		entry.GetName(),
+		entry.GetStackTrace())
+}
+
+// FormatEvent produces a string for a single event in a span that contains the
+// formatted information about that event.  Also used by exporters that
```
Comment (MikeCJohnsonUk) [inline internal/tracing/exporters/common/common.go:84]: Also use by exporters that ...?

File: internal/services/frontend/session_manager.go (hunk 1 above)
Comment (Jim3Things) [reply internal/services/frontend/session_manager.go:78]: This is one of the sequences that the goland linter complains about.  As is the other one below.  Just trying to shut the complaints up...

Comment (MikeCJohnsonUk) [reply internal/services/frontend/session_manager.go:78]: Fairy snuff.

**Pull request merged**